File: Problem_10/main.py
import numpy
from numba import njit


@njit
def get_sum_primes(limit):
    # I am discarding the 2, thus, I need to add it later
    numbers = numpy.arange(3, limit + 1, 2)
    suma = 0
    while len(numbers) > 0:
        suma += numbers[0]
        numbers = numbers[numbers % numbers[0] != 0]

    return suma + 2

def main():
    # run the function in order to compile the function
    get_sum_primes(10)

    # thus, run for the real value [Finished in 75.2s]
    print(get_sum_primes(2000000))

if __name__ == '__main__':
    main()

File: Problem_10/main.cpp
#include <iostream>

int isPrime(int num)
{
    for (int i = 2; i <= num/2; i++)
    {   
        if (num % i == 0)
        {
            return false;
            break;
        }
    }
    return true; 
}


int main(){

    size_t limit = 2000000;
    size_t suma = 0;

    for (int j = 2; j < limit; j++){
        if (isPrime(j)){
            suma += j;
        }
        
    }
    std::cout << suma << std::endl;
    return 0;
}
